machine/machine_core/machine_virtual.py

```diff
@@ -323,13 +323,8 @@ def execute(*args):
         image_to_use = self.image_file
         if not self.maintain:
             (unused, self._transient_image) = tempfile.mkstemp(suffix='.qcow2', prefix='cockpit-', dir=self.run_dir)
-            options = ""
-            # check if it is qcow2 (libvirt complains otherwise)
-            with open(self.image_file, "rb") as f:
-                if f.read(3) == b"QFI":
-                    options += ",backing_fmt=qcow2"
```

Comment:
That's what we had in the past. This was specifically introduced in commit d2ced0c18c23c2d to also support raw overlays for testing lorax. Let's please drop this commit for now; if you care much, please check with @atodorov whether it's still needed, but I don't see much reason to stop supporting it.

Reply:
Reworked this a bit to look nicer.